### ISSUE TITLE
Code generation fails with H2 2.1.214 when using CASE_INSENSITIVE_IDENTIFIERS

### ISSUE THREAD

**Issue by MertCingoz** [jOOQ/jOOQ#13904]
### Expected behavior

To have codegen succeed.

Codegen works as expected with H2 1.4.200 but only by upping H2 2.1.214 it fails. The failure is not our SQL, but the one executed by plugin.

### Actual behavior

Error output is:
```
[INFO] ----------------------------------------------------------
[INFO] Generating catalogs      : Total: 1
[INFO] ARRAYs fetched           : 0 (0 included, 0 excluded)
[INFO] Domains fetched          : 0 (0 included, 0 excluded)
[INFO] Tables fetched           : 33 (33 included, 0 excluded)
[INFO] Embeddables fetched      : 0 (0 included, 0 excluded)
[INFO] Enums fetched            : 0 (0 included, 0 excluded)
[INFO] Packages fetched         : 0 (0 included, 0 excluded)
[INFO] Routines fetched         : 0 (0 included, 0 excluded)
[INFO] Sequences fetched        : 0 (0 included, 0 excluded)
[INFO] No schema version is applied for catalog . Regenerating.
[INFO]                          
[INFO] Generating catalog       : DefaultCatalog.java
[INFO] ==========================================================
[INFO] Generating schemata      : Total: 1
[INFO] No schema version is applied for schema PUBLIC. Regenerating.
[INFO] Generating schema        : DefaultSchema.java
[INFO] ----------------------------------------------------------
[INFO] UDTs fetched             : 0 (0 included, 0 excluded)
[INFO] Generating tables        
[INFO] Generating table         : Audit.java [input=audit, output=audit, pk=N/A]
[WARNING] SQL exception            : Exception while executing meta query: Schema "information_schema" not found; SQL statement:
select INFORMATION_SCHEMA.INDEXES.TABLE_SCHEMA, INFORMATION_SCHEMA.INDEXES.TABLE_NAME, INFORMATION_SCHEMA.INDEXES.INDEX_NAME, (INDEXES.INDEX_TYPE_NAME <> 'UNIQUE INDEX') NON_UNIQUE, i.column_name COLUMN_NAME, i.ordinal_position ORDINAL_POSITION, i.ordering_specification ASC_OR_DESC from INFORMATION_SCHEMA.INDEXES join information_schema.index_columns i on ((INFORMATION_SCHEMA.INDEXES.TABLE_SCHEMA = i.table_schema) and (INFORMATION_SCHEMA.INDEXES.TABLE_NAME = i.table_name) and (INFORMATION_SCHEMA.INDEXES.INDEX_NAME = i.index_name)) where (INFORMATION_SCHEMA.INDEXES.TABLE_SCHEMA in (cast(? as varchar)) and not (INFORMATION_SCHEMA.INDEXES.IS_GENERATED)) order by INFORMATION_SCHEMA.INDEXES.TABLE_SCHEMA, INFORMATION_SCHEMA.INDEXES.TABLE_NAME, INFORMATION_SCHEMA.INDEXES.INDEX_NAME, i.ordinal_position [90079-214]
If you think this is a bug in jOOQ, please report it here: https://github.com/jOOQ/jOOQ/issues/new
```

### Steps to reproduce the problem

I am getting very similar error with #13240,
H2 error Schema "information_schema" not found (uses lowercase information_schema)

Database url:
... ;DATABASE_TO_UPPER=FALSE;CASE_INSENSITIVE_IDENTIFIERS=TRUE

It fails on this query:
https://github.com/jOOQ/jOOQ/blob/version-3.16.0/jOOQ-meta/src/main/java/org/jooq/meta/h2/H2Database.java#L249

### Versions

- jOOQ: 3.16.9
- Java: Temurin-11.0.16+8
- Database (include vendor): H2 versions 1.4.200 and 2.1.214
- OS: macOS
- JDBC Driver (include name if unofficial driver): H2

**Comment by katzyn** [comment]
`DATABASE_TO_UPPER=FALSE` is suitable only for H2 1.4.197 and older versions.

In H2 1.4.198 Beta and newer versions `DATABASE_TO_LOWER=TRUE` should be used instead of it with MariaDB, MySQL, and PostgreSQL compatibility modes. `CASE_INSENSITIVE_IDENTIFIERS=TRUE` may additionally be used with `MariaDB` and `MySQL` compatibility modes in some cases, but usually should be avoided. It doesn't affect case sensitivity of schema names.

@lukaseder 
If jOOQ needs to support H2 with these strange settings, `INFORMATION_SCHEMA` should be specified in upper case in queries, this variant is expected to work with all combinations of settings.

**Comment by lukaseder** [comment]
Thanks for your report, @MertCingoz 

> If jOOQ needs to support H2 with these strange settings, `INFORMATION_SCHEMA` should be specified in upper case in queries, this variant is expected to work with all combinations of settings.

jOOQ doesn't support these settings, but the fact that the `information_schema` identifier was written in lower case in some plain SQL template from `jOOQ-meta` is a minor bug. It doesn't affect 99% of all users, but there's always this edge case. 😅

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.4 (https://github.com/jOOQ/jOOQ/issues/13909), and 3.16.10 (https://github.com/jOOQ/jOOQ/issues/13910)
